cmd/snap/inhibit.go

```diff
@@ -34,143 +34,153 @@ import (
-		return false, nil
+		return false
 	}
 	obj := conn.Object("io.snapcraft.SnapDesktopIntegration", "/io/snapcraft/SnapDesktopIntegration")
 	extraParams := make(map[string]dbus.Variant)
 	err = obj.Call("io.snapcraft.SnapDesktopIntegration.ApplicationIsBeingRefreshed", 0, snapName, runinhibit.HintFile(snapName), extraParams).Store()
 	if err != nil {
 		logger.Noticef("unable to successfully call io.snapcraft.SnapDesktopIntegration.ApplicationIsBeingRefreshed: %v", err)
-		return false, nil
+		return false
 	}
-	if _, err := waitInhibitUnlock(snapName, runinhibit.HintNotInhibited); err != nil {
-		return false, err
-	}
-	return true, nil
+	return true
 }
 
-func graphicalSessionFlow(snapName string, hint runinhibit.Hint) error {
-	refreshInfo := client.PendingSnapRefreshInfo{
-		InstanceName: snapName,
-		// Remaining time = 0 results in "Snap .. is refreshing now" message from
-		// usersession agent.
-		TimeRemaining: 0,
-	}
+var isGraphicalSession = func() bool {
+	// TODO: uncomment once there is a proper UX review
+	//return os.Getenv("DISPLAY") != "" || os.Getenv("WAYLAND_DISPLAY") != ""
+	return false
+}
 
-	if err := pendingRefreshNotification(&refreshInfo); err != nil {
-		return err
-	}
-	if _, err := waitInhibitUnlock(snapName, runinhibit.HintNotInhibited); err != nil {
+var pendingRefreshNotification = func(refreshInfo *client.PendingSnapRefreshInfo) error {
+	userclient := client.NewForUids(os.Getuid())
+	if err := userclient.PendingRefreshNotification(context.TODO(), refreshInfo); err != nil {
 		return err
 	}
-
-	finishRefreshInfo := client.FinishedSnapRefreshInfo{InstanceName: snapName}
-	return finishRefreshNotification(&finishRefreshInfo)
-}
-
-func textFlow(snapName string, hint runinhibit.Hint) error {
-	fmt.Fprintf(Stdout, "%s\n", inhibitMessage(snapName, hint))
-	pb := progress.MakeProgressBar(Stdout)
-	pb.Spin(i18n.G("please wait..."))
-	_, err := waitInhibitUnlock(snapName, runinhibit.HintNotInhibited)
-	pb.Finished()
-	return err
+	return nil
 }
 
-var isLocked = runinhibit.IsLocked
-
-// waitInhibitUnlock waits until the runinhibit lock hint has a specific waitFor value
-// or isn't inhibited anymore.
-var waitInhibitUnlock = func(snapName string, waitFor runinhibit.Hint) (notInhibited bool, err error) {
-	// Every 0.5s check if the inhibition file is still present.
-	ticker := time.NewTicker(500 * time.Millisecond)
-	defer ticker.Stop()
-	for {
-		select {
-		case <-ticker.C:
-			// Half a second has elapsed, let's check again.
-			hint, _, err := isLocked(snapName)
-			if err != nil {
-				return false, err
-			}
-			if hint == runinhibit.HintNotInhibited {
-				return true, nil
-			}
-			if hint == waitFor {
-				return false, nil
-			}
-		}
+var finishRefreshNotification = func(refreshInfo *client.FinishedSnapRefreshInfo) error {
+	userclient := client.NewForUids(os.Getuid())
+	if err := userclient.FinishRefreshNotification(context.TODO(), refreshInfo); err != nil {
```

Comment:
I would say that the context should be in the interface so the proper context can be provided and controlled by the caller.